ada_feeding/ada_feeding/visitors/moveto_visitor.py

```diff
@@ -172,18 +172,10 @@ def run(self, behaviour: py_trees.behaviour.Behaviour) -> None:
         ):
             # Flip to execute
             if self.feedback.is_planning:
-                self.start_time = self.node.get_clock().now()
+                self.start_time = behaviour.start_time
```

Comment:
I would update this class as follows:

(1) Keep a reference to the running Execute behavior (e.g. ServoMove, MoveIt2Execute) and only execute the IsPlanning logic (i.e. under `else` below on Line 179) when that running Execute behavior returns Success/Failure. This handles sequences/selectors w/o memory so we don't have to worry about planning/execute switching.

(2) Isolate the motion initial/current distance logic in here so it doesn't pollute the tree.

Look for an ApplyTransform returning SUCCESS with the linear distance output.

If we are planning, set it to motion initial distance (it doesn't matter if we give "garbage" values in motion if we are planning, since by spec those values are meaningless if we are planning anyway). If we are executing, set it to motion current distance.

Now you can remove the caching logic from ServoMove, which isn't relevant to that behavior anyway.